pkg/util/cluster.go

```diff
@@ -56,32 +108,39 @@ func CreateClusterObject(controlPlaneClient *karmadaclientset.Clientset, cluster
 
 // CreateOrUpdateClusterObject create cluster object in karmada control plane,
 // if cluster object has been existed and different from input clusterObj, update it.
-func CreateOrUpdateClusterObject(controlPlaneClient *karmadaclientset.Clientset, clusterObj *clusterv1alpha1.Cluster, mutate func(*clusterv1alpha1.Cluster)) (*clusterv1alpha1.Cluster, error) {
+func CreateOrUpdateClusterObject(controlPlaneClient karmadaclientset.Interface, clusterObj *clusterv1alpha1.Cluster, mutate func(*clusterv1alpha1.Cluster)) (*clusterv1alpha1.Cluster, error) {
 	cluster, exist, err := GetClusterWithKarmadaClient(controlPlaneClient, clusterObj.Name)
 	if err != nil {
 		return nil, err
 	}
-
-	if exist {
-		if reflect.DeepEqual(cluster.Spec, clusterObj.Spec) {
-			klog.Warningf("cluster(%s) already exist and newest", clusterObj.Name)
+	if mutate == nil {
```

Comment:
This mutate nil judgment can be removed.

Reply:
> When report-secrets is {"KubeCredentials", "KubeImpersonator"}, only impersonator secret will be report.

That's truly a problem, i fix it, thanks~

Reply:
@XiShanYongYe-Chang  why fetch image failed